lib/runner/index.js

```diff
@@ -101,35 +104,39 @@ _.assign(Runner, {
      * @private
      *
      * @param {?Collection} collection
-     * @param {?String} groupId
+     * @param {?String} entrypoint folder/request id or name to begin extraction
      * @param {Function} callback
      * @returns {Object}
      * @todo Add support for accepting groupId as an array of ids/names
      */
-    extractRunnableItems: function (collection, groupId, callback) {
-        var idMatch,
-            nameMatch,
-            items = [],
-            group = !groupId && collection; // if a group id is unspecified, the collection is our group
-
-        // if group id is not provided, we try and check if we can find it in the collection
-        if (!group) {
-            collection && collection.forEachItemGroup && collection.forEachItemGroup(function (itemGroup) {
-                // Matching has been done like this to avoid having to go over the collection item groups twice
-                (itemGroup.id === groupId) && (idMatch = itemGroup);
-                (itemGroup.name === groupId) && (nameMatch = itemGroup);
-            });
-
-            group = idMatch || nameMatch; // Prioritise id based matches over name based ones.
+    extractRunnableItems: function (collection, entrypoint, callback) {
+        var items = [],
+            entryNode;
+
+        // if an entrypoint is specified, find the item/item group
+        // looks for first node with matching id or name
+        if (entrypoint) {
+            // @todo: fix this. the check by id should all finish before doing check by name
+            collection && collection.oneDeep && (entryNode = collection.oneDeep(entrypoint));
         }
 
-        // accumulate all items in the collection/group in order
-        // @todo: Possibly add mapItem to sdk.ItemGroup?
-        group && group.forEachItem && group.forEachItem(function (item) {
-            items.push(item);
-        });
+        // no entry point specified, start with the collection
+        else { entryNode = collection; }
+
+        // entry point lookup did not find anything
+        if (!entryNode) {
+            callback(null, items);
+            return;
+        }
+
+        // accumulate all items in order if entry point is a collection/folder
+        ItemGroup.isItemGroup(entryNode) ?
```

Comment:
This can be made easier to read, the behaviour of prioritizing folder matches over requests should also be documented.

Reply:
This is not prioritizing folder matches, a node can only be `Item` or `ItemGroup`, but not both.

Reply:
In collections that have folders and requests with the same name, this will become a problem, but that will get resolved at the `oneDeep` call itself.